app/models/exercise.rb

```diff
@@ -96,7 +96,7 @@ def self.for(track_slug, exercise_slug)
   after_update_commit do
     if saved_changes.include?(:git_important_files_hash)
       Exercise::MarkSolutionsAsOutOfDateInIndex.defer(self)
-      Exercise::QueueSolutionHeadTestRuns.defer(self)
+      Exercise::QueueSolutionHeadTestRuns.defer(self) unless skip_test_runs?
```

Comment:
I think I'd like to move this into `Exercise::QueueSolutionHeadTestRuns` itself. It feels better encapsulated to have the logic there, in case we end up running a batch script later etc.